sonar-orchestrator/src/main/java/com/sonar/orchestrator/container/Server.java

```diff
@@ -204,18 +227,86 @@ public SonarClient adminWsClient() {
+          .setAdminCredentials()
+          .setParam("backup", IOUtils.toString(input, UTF_8))
+          .execute();
+      } else {
+        restoreProfileForOldVersion(new ByteArrayBody(IOUtils.toByteArray(input), ContentType.TEXT_XML, "profile-backup.xml"));
+      }
       return this;
     } catch (IOException e) {
       throw new IllegalStateException("Cannot restore Quality profile", e);
     }
   }
 
+  private Server restoreProfileForOldVersion(ContentBody backup) {
+    if (url == null) {
+      throw new IllegalStateException("Can not restore profiles backup if the server is not started");
+    }
+    // still no sonar-ws-client for this web service
+    DefaultHttpClient client = new DefaultHttpClient();
+    try {
+      HttpHost targetHost = new HttpHost("localhost", new URL(getUrl()).getPort(), "http");
+
+      HttpParams params = client.getParams();
+      HttpConnectionParams.setConnectionTimeout(params, 60_000);
+      HttpConnectionParams.setSoTimeout(params, 120_000);
+
+      client.getCredentialsProvider().setCredentials(new AuthScope(targetHost.getHostName(), targetHost.getPort()),
+        new UsernamePasswordCredentials(ADMIN_LOGIN, ADMIN_PASSWORD));
+
+      BasicHttpContext localcontext = new BasicHttpContext();
+      BasicScheme basicAuth = new BasicScheme();
+
+      localcontext.setAttribute("preemptive-auth", basicAuth);
+      client.addRequestInterceptor(new PreemptiveAuth(), 0);
+
+      String wsUrl = url + "/api/qualityprofiles/restore";
+      HttpPost post = new HttpPost(wsUrl);
+      MultipartEntity entity = new MultipartEntity(HttpMultipartMode.BROWSER_COMPATIBLE);
+      entity.addPart("backup", backup);
+      post.setEntity(entity);
+      HttpResponse httpResponse = client.execute(post, localcontext);
+      int statusCode = httpResponse.getStatusLine().getStatusCode();
+
+      // Versions less than 3.1 request a standard HTTP/HTML service, but not the web service.
+      // For this reason we still check the status 302.
+      if (statusCode != 200 && statusCode != 302) {
+        throw new IllegalStateException("Fail to restore profile backup, status: " + httpResponse.getStatusLine());
+      }
+      return this;
+
+    } catch (IOException e) {
+      throw new IllegalStateException("Fail to restore profile backup", e);
+
+    } finally {
+      client.close();
+    }
+  }
+
+  static final class PreemptiveAuth implements HttpRequestInterceptor {
+    @Override
+    public void process(
+      final HttpRequest request,
+      final HttpContext context) throws HttpException {
+
+      AuthState authState = (AuthState) context.getAttribute(ClientContext.TARGET_AUTH_STATE);
+
+      if (authState.getAuthScheme() == null) {
+        AuthScheme authScheme = (AuthScheme) context.getAttribute("preemptive-auth");
+        if (authScheme != null) {
+          authState.setAuthScheme(authScheme);
+          authState.setCredentials(new UsernamePasswordCredentials(ADMIN_LOGIN, ADMIN_PASSWORD));
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Remove this use of "setCredentials"; it is deprecated. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3ACallToDeprecatedMethod)